tests/generators/runners/ssz_generic_cases/ssz_compatible_union.py

```diff
@@ -0,0 +1,211 @@
+    ProgressiveVarTestStruct,
+)
+from .ssz_test_case import invalid_test_case, valid_test_case
+
+CompatibleUnionA = CompatibleUnion({1: ProgressiveSingleFieldContainerTestStruct})
+
+CompatibleUnionBC = CompatibleUnion(
+    {2: ProgressiveSingleListContainerTestStruct, 3: ProgressiveVarTestStruct}
+)
+
+CompatibleUnionABCA = CompatibleUnion(
+    {
+        1: ProgressiveSingleFieldContainerTestStruct,
+        2: ProgressiveSingleListContainerTestStruct,
+        3: ProgressiveVarTestStruct,
+        4: ProgressiveSingleFieldContainerTestStruct,
+    }
+)
+
+PRESET_COMPATIBLE_UNIONS: dict[str, type[View]] = {
+    "CompatibleUnionA": CompatibleUnionA,
+    "CompatibleUnionBC": CompatibleUnionBC,
+    "CompatibleUnionABCA": CompatibleUnionABCA,
+}
+
+
+def valid_compatible_union_cases(rng: Random, name: str, typ: type[View]):
+    for option, elem_type in typ.options().items():
+        for mode in [RandomizationMode.mode_zero, RandomizationMode.mode_max]:
+            yield (
+                f"{name}_{mode.to_name()}_selector_{option}",
+                valid_test_case(
+                    lambda rng=rng,
+                    mode=mode,
+                    typ=typ,
+                    elem_type=elem_type,
+                    option=option: deserialize(
+                        typ,
+                        bytes([option]) + serialize(container_case_fn(rng, mode, elem_type)),
+                    )
+                ),
+            )
+        for mode in list(RandomizationMode):
+            for variation in range(3):
+                yield (
+                    f"{name}_{mode.to_name()}_selector_{option}_chaos_{variation}",
+                    valid_test_case(
+                        lambda rng=rng,
+                        mode=mode,
+                        typ=typ,
+                        elem_type=elem_type,
+                        option=option: deserialize(
+                            typ,
+                            bytes([option])
+                            + serialize(container_case_fn(rng, mode, elem_type, chaos=True)),
+                        )
+                    ),
+                )
+            if mode == RandomizationMode.mode_random:
+                for variation in range(10):
+                    yield (
+                        f"{name}_{mode.to_name()}_selector_{option}_{variation}",
+                        valid_test_case(
+                            lambda rng=rng,
+                            mode=mode,
+                            typ=typ,
+                            elem_type=elem_type,
+                            option=option: deserialize(
+                                typ,
+                                bytes([option])
+                                + serialize(container_case_fn(rng, mode, elem_type)),
+                            )
+                        ),
+                    )
+
+
+def valid_cases():
+    rng = Random(1234)
+    for name, typ in PRESET_COMPATIBLE_UNIONS.items():
```

Comment:
This is a dictionary & we need to iterate over these in a deterministic way. Otherwise the reference tests may be different between runs. Same thing applies to the invalid cases.